addons/website/static/src/js/content/redirect.js

```diff
@@ -16,62 +14,24 @@ document.addEventListener('DOMContentLoaded', () => {
     }
 
     if (!window.frameElement) {
-        const websiteId = document.documentElement.dataset.websiteId;
-        const {pathname, search} = window.location;
-        const params = new URLSearchParams(search);
-        const enableEditor = params.get('enable_editor');
-        const editTranslations = params.get('edit_translations');
-        params.delete('enable_editor');
-        let newSearch = params.toString();
-        if (newSearch) {
-            newSearch = '?' + newSearch;
+        const frontendToBackendNavEl = document.querySelector('.o_frontend_to_backend_nav');
+        if (frontendToBackendNavEl) {
+            frontendToBackendNavEl.classList.add('d-flex');
+            frontendToBackendNavEl.classList.remove('d-none');
         }
-        const backendPath = `/web#action=website.website_preview&path=${encodeURIComponent(`${pathname}${newSearch}`)}&website_id=${websiteId}`;
 
-        const autoredirectToBackendAction = enableEditor || editTranslations;
-        if (autoredirectToBackendAction) {
+        // Auto redirect to frontend if edit/translation mode is requested
+        const currentUrl = new URL(window.location.href);
+        currentUrl.pathname = `/@${currentUrl.pathname}`;
+        const href = currentUrl.href;
```

Comment:
Can't this be done server side as well in the end?

Reply:
No I am talking about rendering the UI buttons in your XML template with the right URL directly, instead of letting JS code do it.

Reply:
This is not possible as the anchor are not sent to the server, this is something only kept in the browser and that only the browser knows, so we can't generate the URL server side / XML side sadly, or it would be missing those (important) anchors.

Reply:
Well, exactly :grimacing: Are not they lost already in any case? :grimacing:

Your code:
- Load url `/foo/bar#hello=world`
- In JS, create the href `/@/foo/bar#hello=world` and set it to the DOM button
- The user clicks on it, the `/@/` controller redirects to ~ `/web?action=website.website_preview&path=/foo/bar` -> the hash is lost

What I suggested:
- Load url `/foo/bar#hello=world`, the rendered XML contains a button to `/@/foo/bar` -> the hash is lost
- The user clicks on it, the `/@/` controller redirects to ~ `/web?action=website.website_preview&path=/foo/bar`